client/table.go

```diff
@@ -287,11 +287,11 @@ func (db *DB) CreateTable(desc *structured.TableDescriptor) error {
 		return fmt.Errorf("table \"%s\" already exists", desc.Name)
 	}
 
-	ir, err := db.Inc(keys.DescIDGenerator, 1)
-	if err != nil {
+	if ir, err := db.Inc(keys.DescIDGenerator, 1); err == nil {
+		desc.ID = uint32(ir.ValueInt() - 1)
+	} else {
```

Comment:
This version placates both linters, but violates the more general pattern that error cases are handled first. For an isolated case it doesn't matter too much, but personally I'd rather suppress this warning from golint than reorder clauses throughout the codebase to satisfy it.